### PR TITLE
feat(insights): Update backend query cache staleness

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

TLDR: Async dashboards trigger query calculations too often (mins or hrs). Current dashboard auto trigger (in frontend) is 20h.

We are improving the async behavior of dashboards to ultimately show fresher data on dashboards both normal and shared ones.
There is a feature flagged "async dashboard" loading, which triggers calculations on the backend according to staleness.
However, the existing frontend logic was to trigger auto refresh when the cache is 20 hours old (albeit users just clicked the refresh button when they want).
Considering the backend staleness deems an insight stale after a few minutes or hours, this is a pretty a large difference that leads to quite some more calculations when rolling out the async dashboard more.

## Changes

- align staleness across runners: they all used the same logic
- pin down one set of staleness thresholds and cover with test
- extend the "lazy" cache/refresh logic: if lazy, we use extended cache lifetimes. This can basically replace the old "just wait 20 hours" in the frontend after we remove the feature flag.
- both internal and shared dashboards will be lazy, thus auto trigger async refreshes when deemed stale this way
- this bridges the gap from the current "20 hours" behavior to a new lazy async behavior

### Future plans

- roll out feature flag
- remove refresh code from the dashboard logic - we can just ask the dashboard endpoint again - only polling remains
- remove the update cached task layer - it's not working and the async stuff here replaces it
- make individual shared insights also load with the lazy cache async mode -> needs polling though, or results won't show

## Does this work well for both Cloud and self-hosted?

n/a

## How did you test this code?

Added tests
Checked dashboard and query behavior

### CODE REVIEW

**Review by thmsobrmlr — APPROVED** [review]
Good progress on improving the code, all with cleanups and tests, thanks!
